database/views.py

```diff
@@ -139,14 +140,31 @@ def get_context_data(self, **kwargs):
         context['reaction'] = reaction
         context['reactants'] = reaction.reactants()
         context['products'] = reaction.products()
+        try:
+            context['kinetics'] = reaction.kinetics_set.all()
+        except:
+            context['kinetics'] = None
         return context
 
 class KineticsDetail(DetailView):
-    model = BaseKineticsData
+    model = Kinetics
 
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
         kinetics = self.get_object()
-        #kinetic_model = KineticModel.objects.get(source=source)
-        context['kinetics'] = kinetics
+        kineticdata = None
+        for kin_type in ['arrhenius', 'arrheniusep', 'chebyshev', 'lindemann', 'multiarrhenius', 'multipdeparrhenius', 'pdeparrhenius', 'thirdbody', 'troe']:
+            try:
+                kineticdata = eval(f'kinetics.basekineticsdata.{kin_type}')
+                break
+            except:
+                continue
+        try:
+            context['kinetics'] = kinetics
+            context['kin_type'] = kin_type
+            context['kineticdata'] = kineticdata
+        except: 
+            context['kinetics'] = None
+            context['kin_type'] = None
+            context['kineticdata'] = None
```

Comment:
A couple things here:
1. Won't `kin_type` always be 'troe'?
2. Do we need this block? `kinetics` is already passed to the context with the name `object` since we are using a `DetailView`. `kineticdata` is also initialized as `None` so the assignment in the `try` should cover all the cases.

Reply:
For 1: I see that works now since it's `break`ing. I still think it would be better to use another variable so that the block addressed in 2 can be removed. I can make these changes if you want!

Reply:
Go ahead and make these changes!

Reply:
dfe1c06